doc/man3/OPENSSL_LH_COMPFUNC.pod

```diff
@@ -184,6 +184,13 @@ audit/verify and also opens the window of opportunity for stack
 corruption and other hard-to-find bugs.  It also, apparently, violates
 ANSI-C.
 
+The LHASH code is not thread safe. All updating operations must be
+performed under a write lock. All retrieve operations should be performed
+under a read lock, I<unless> accurate usage statistics are desired.
+In which case, a write lock should be used for retreve operations as well.
+For output of usage statistics, using the functions from L<OPENSSL_LH_stats(3)>,
+a read lock always suffices.
```

Comment:
I'm not exactly happy about "always". Even though we can't deny user looking at inaccurate statistics, we probably shouldn't encourage it. I mean I feel that last sentence effectively does the latter. But on the other hand it might as well happen that I'm the only one who gets that impression. In other words I'd omit "always", would anybody else? I also wonder what effect would "the" have in "For output of *the* usage statistics". Would it be interpreted rather as "for output of [abovementioned accurate] usage statistics"? If so, it would also be an improvement in my ears...

Reply:
Both suggestions incorporated and squashed.